manifest/oci_index.go

```diff
@@ -89,11 +98,20 @@ func (index *OCI1Index) ChooseInstance(ctx *types.SystemContext) (digest.Digest,
 				Variant:      d.Platform.Variant,
 			}
 			if platform.MatchesPlatform(imagePlatform, wantedPlatform) {
-				return d.Digest, nil
+				matchedDigest = d.Digest
+				// Always give priority to zstd if more than one image
+				// exists for the same platform.
+				if _, ok := d.Annotations[ZSTDInstanceAnnotation]; ok {
+					return matchedDigest, nil
+				}
 			}
 		}
 	}
 
+	if matchedDigest != "" {
+		return matchedDigest, nil
+	}
+
 	for _, d := range index.Manifests {
```

Comment:
Shouldn’t we use the same logic here as well?

That probably suggests defining some kind of `instanceBetterMatchesPlatforms` predicate (which can be unit-tested separately), and this function reducing to a single loop that finds the best entry.